components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -468,6 +468,33 @@ mod parser {
             _ => None,
         }
     }
+
+    /// Try to parse a `u64` into a `Time` with the given type and fsp
+    pub fn parse_from_u64(
+        ctx: &mut EvalContext,
+        input: u64,
+        time_type: TimeType,
+        fsp: u8,
+    ) -> Option<Time> {
+        if input == 0 {
+            return Time::zero(ctx, fsp as i8, time_type).ok();
+        }
+        // NOTE: These numbers can be consider as strings
+        // The parser eats two digits each time from the end of string,
+        // and fill it into `Time` with reversed order.
+        // Port from: https://github.com/pingcap/tidb/blob/b1aad071489619998e4caefd235ed01f179c2db2/types/time.go#L1263
+        let aligned = match input {
```

Comment:
What about the input between 991231235959 and 10000101000000

Reply:
Because of any integer in this range will not consitute a valid date, we can just reture `None` while we encounter this range.